src/gossip/gossip_service.zig

```diff
@@ -1528,6 +1543,37 @@ pub const GossipService = struct {
         }
     }
 
+    /// Attempts to associate each entrypoint address with a contact info.
+    /// Returns true if all entrypoints have been identified
+    fn identifyEntrypoints(self: *Self) bool {
+        var identified_all = true;
+        for (self.entrypoints.items) |*entrypoint| {
+            if (entrypoint.info == null) {
+                var reader = self.crds_table_rw.read();
+                defer reader.unlock();
+                entrypoint.info = reader.get().getContactInfoByGossipAddr(entrypoint.addr);
+            }
+            identified_all = identified_all and entrypoint.info == null;
+        }
+        return identified_all;
+    }
+
+    /// if we have no shred version, attempt to get one from an entrypoint.
+    /// Returns true if the shred version is set to non-zero
+    fn assignDefaultShredVersionFromEntrypoint(self: *Self) bool {
+        if (self.my_shred_version.load(.Monotonic) != 0) return true;
+        for (self.entrypoints.items) |entrypoint| {
+            if (entrypoint.info) |info| {
+                if (info.shred_version != 0) {
+                    self.my_shred_version.store(info.shred_version, .Monotonic);
```

Comment:
probs wana log the shred version change here

Reply:
https://github.com/Syndica/sig/pull/69/commits/06eed7ccf099815ae74f96ae1d0cf154ed4b569c

I also added some information about the source of the shred version, since it can come from multiple places.

I realized there are a lot of ways to return a string from a function. It's not clear to me that there's a nice clean idiomatic approach for this in zig. I aimed for something simple that avoids an allocation, since we know the maximum string size in this case is not very big.

Reply:
yeah im not sure how we could do better so i think its ok